batch/testResultsCollector.js

```diff
@@ -2,39 +2,73 @@ import {XUnitParser} from './parsers/xUnitParser';
 import mongoose from 'mongoose';
 import {TestSuiteModel} from '../models/testSuiteModel';
 import {TestCaseModel} from '../models/testCaseModel';
-
+import {GithubHelper} from '../helpers/githubHelper';
 
 class TestResultsCollector {
 	constructor(configuration) {
 		this.configuration = configuration;
-		mongoose.connect(configuration.mongo.uri, configuration.mongo.options);
+	}
+
+	connect() {
+		mongoose.connect(this.configuration.mongo.uri, this.configuration.mongo.options);
+	}
+
+	get parser() {
+		return new XUnitParser(this.configuration);
```

Comment:
Why return a new one on each access? Especially since it's a getter: this means `collector.parser !== collector.parser`. It's especially meaningless since the configuration is set at instantiation.